src/baseTrie.ts

```diff
@@ -669,37 +733,30 @@ module.exports = class Trie {
   }
 
   // Creates the initial node from an empty tree
-  _createInitialNode (key, value, cb) {
+  _createInitialNode(key: Buffer, value: Buffer, cb: Function) {
     const newNode = new TrieNode('leaf', key, value)
     this.root = newNode.hash()
     this._putNode(newNode, cb)
   }
 
   // formats node to be saved by levelup.batch.
   // returns either the hash that will be used key or the rawNode
-  _formatNode (node, topLevel, remove, opStack) {
-    if (arguments.length === 3) {
-      opStack = remove
-      remove = false
-    }
-
+  _formatNode(
+    node: TrieNode,
+    topLevel: boolean,
+    opStack: BatchDBOp[],
+    remove: boolean = false,
+  ): Buffer | Buffer[] {
     const rlpNode = node.serialize()
 
     if (rlpNode.length >= 32 || topLevel) {
       const hashRoot = node.hash()
 
-      if (remove && this.isCheckpoint) {
-        opStack.push({
-          type: 'del',
-          key: hashRoot
-        })
-      } else {
-        opStack.push({
-          type: 'put',
-          key: hashRoot,
-          value: rlpNode
-        })
-      }
```

Comment:
The reason I removed this was because I thought `BaseTrie` should not have `CheckpointTrie` specific logic. `this.isCheckpoint` is only defined in the `CheckpointTrie`. So I duplicated the `_formatNode` function in `CheckpointTrie` and moved this logic there.